### PR TITLE
Add environment variables OPAMASSUMEDEPEXTS, OPAMNODEPEXTS

### PR DESCRIPTION
**Pull request by AltGr**
Split out from #4504

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -254,7 +254,7 @@ AC_CHECK_TOOL(CPPO,cppo)
 AC_CHECK_TOOL(PATCH,patch)
 AC_CHECK_TOOL(BUNZIP2,bunzip2)
 
-AX_COMPARE_VERSION([$OCAMLVERSION], [lt], [4.07.0],
+AX_COMPARE_VERSION([$OCAMLVERSION], [lt], [4.08.0],
```
Comment (dra27) [inline configure.ac:257]: Ouch (my mistake...) 😕

**Review by dra27 — APPROVED** [review]
LGTM!

**Review by rjbou — APPROVED** [review]

File: configure.ac (hunk 1 above)
Comment (rjbou) [reply configure.ac:257]: `configure` not updated accordingly (as for #4592). Shouldn't CI error in that case?

**Review by rjbou — CHANGES_REQUESTED** [review]

**Comment by rjbou** [comment]
Except configure, LGTM! thanks!

File: configure.ac (hunk 1 above)
Comment (dra27) [reply configure.ac:257]: Eek: https://github.com/ocaml/opam/pull/4587/checks?check_run_id=2056831138#step:4:48

Comment (rjbou) [reply configure.ac:257]: ok...

**Comment by rjbou** [comment]
Updated!

**Review by rjbou — APPROVED** [review]

**Pull request merged**